Core/src/org/sleuthkit/autopsy/datamodel/DataSourceGroupingNode.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Autopsy Forensic Browser
+ * 
+ * Copyright 2018 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.datamodel;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.logging.Level;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.autopsy.casemodule.NoCurrentCaseException;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.datamodel.DataSource;
+import org.sleuthkit.datamodel.Image;
+import org.sleuthkit.datamodel.LocalFilesDataSource;
+
+
+/**
+ *  Data source grouping node - an optional grouping node in the data tree view
+ * 
+ */
+public class DataSourceGroupingNode extends DisplayableItemNode {
+
+    private static final Logger logger = Logger.getLogger(DataSourceGroupingNode.class.getName());
+
+    /**
+     * Creates the Data source grouping node for the given data source,  
+     * 
+     * @param dataSource specifies the data source 
+     */
```

Comment:
This comment for the published, public API documentation is a little funky with respect to capitalization and punctuation.